tests/test_network.py

```diff
@@ -124,6 +134,159 @@ def test_add_objective(factory, client):
     assert dataset.objective_key == objective.key
 
 
+@pytest.mark.parametrize('asset_name', [
+    'objective',
+    'algo',
+    'aggregate_algo',
+    'composite_algo',
+])
+def test_metadata(factory, client, asset_name):
+    method_create = getattr(factory, f"create_{asset_name}")
+    method_add = getattr(client, f"add_{asset_name}")
+
+    # add an asset with metadata
+    spec = method_create(metadata={"foo": "bar"})
+    asset = method_add(spec)
+
+    assert asset.metadata == {"foo": "bar"}
+
+    # add asset with metadata set to None
+    spec = method_create(metadata=None)
+    asset = method_add(spec)
+
+    assert asset.metadata == {}
+
+    # add asset with metadata set to an empty dict
+    spec = method_create(metadata={})
+    asset = method_add(spec)
+
+    assert asset.metadata == {}
+
+    spec = method_create(metadata='foo')
+    asset = method_add(spec)
+
+    assert asset.metadata == {}
+
+    spec = method_create(metadata={'foo': {"bar": "foo"}})
+    with pytest.raises(ValueError):
+        method_add(spec)
+
+    too_long = 'foo' * 40
+    # raise an error if the key is too long (more than 50 chars)
+    spec = method_create(metadata={too_long: "bar"})
+    with pytest.raises(substra.exceptions.InvalidRequest):
+        method_add(spec)
+
+    # raise an error if the value is too long (more than 100 chars)
+    spec = method_create(metadata={"foo": too_long})
+    with pytest.raises(substra.exceptions.InvalidRequest):
+        method_add(spec)
```

Comment:
I feel like all the errors should be handled in the same way: always raise an InvalidRequest. Do you think it would be possible to update the backend and sdk code so that we have a consistent behavior from the user standpoint?